### PR TITLE
gtk3-nocsd: init at 3.0.1

### PR DESCRIPTION
**Pull request by Abbie5**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by worldofpeace** [comment]
I'm pretty sure there are other ways to disable CSD to enable SSD in gtk than this.
Not to mention, I'm not really fond of having something in the official nixpkgs repository that's purely a "hack". Also a hack that hasn't seen contributions in 3 or so years.
I do see other distro's have packaged it
* https://repology.org/project/gtk3-nocsd/versions

but it seems to one of those things that might already be abandoned, and thusly if I were to merge this it would add to the abondonware we already have in nixpkgs.

**Comment by peanutbutter144** [comment]
If there are other ways to disable CSD in GTK 3 I haven't heard of them.

There is an [active fork][1] of gtk3-nocsd with a few fixes, so I can package that instead if that would be better.

[1]: https://github.com/ZaWertun/gtk3-nocsd

**Comment by worldofpeace** [comment]
@peanutbutter144 Yep, active forks are pretty much the only way I think this could be acceptable.

File: pkgs/development/libraries/gtk3-nocsd/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+, gobject-introspection
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gtk3-nocsd";
+  version = "3.0.1";
+
+  src = fetchFromGitHub {
+    owner = "ZaWertun";
+    repo = "${pname}";
```
Comment (worldofpeace) [inline pkgs/development/libraries/gtk3-nocsd/default.nix]:
```suggestion
    repo = pname;
```

File: pkgs/development/libraries/gtk3-nocsd/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    sha256 = "035rrn9jq9bdfkmmj6xl4q8paqx7xf3hxsw6gslgk86sh7x56lvi";
+  };
+
+  nativeBuildInputs = [
+    pkgconfig
+    gtk3
+    gobject-introspection
+  ];
+
+  makeFlags = [ "prefix=$(out)" ];
```
Comment (worldofpeace) [inline pkgs/development/libraries/gtk3-nocsd/default.nix]:
```suggestion
  makeFlags = [ 
    "prefix=${placeholder "out"}" 
  ];
```

File: pkgs/development/libraries/gtk3-nocsd/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  version = "3.0.1";
+
+  src = fetchFromGitHub {
+    owner = "ZaWertun";
+    repo = "${pname}";
+    rev = "v${version}";
+    sha256 = "035rrn9jq9bdfkmmj6xl4q8paqx7xf3hxsw6gslgk86sh7x56lvi";
+  };
+
+  nativeBuildInputs = [
```
Comment (worldofpeace) [inline pkgs/development/libraries/gtk3-nocsd/default.nix]: `pkgconfig` should be `native` but the rest should be `buildInputs` because we actually link against them.

Comment (peanutbutter144) [reply pkgs/development/libraries/gtk3-nocsd/default.nix]: Applied

File: pkgs/development/libraries/gtk3-nocsd/default.nix (hunk 2 above)
Comment (peanutbutter144) [reply pkgs/development/libraries/gtk3-nocsd/default.nix]: Applied

**Comment by peanutbutter144** [comment]
@worldofpeace are these changes acceptable to merge?

**Comment by worldofpeace** [comment]
Did you close this due to my non-response @peanutbutter144? My apologies if that caused you to not want to pursue this. nixpkgs is kinda busy, and my notifications are rather hard to parse.

Do you still want to maintain this?

**Comment by peanutbutter144** [comment]
I closed it because 1. I don't use the library any more and 2. I don't want to maintain it.

Sorry for closing without explanation.

**Comment by worldofpeace** [comment]
Thanks for explaining @peanutbutter144 :+1:
